### PR TITLE
bugfix() Fix how "Some::filterIsOneOf()" traverses through received parameters array

### PR DESCRIPTION
**Pull request by mister-spock**

### CODE REVIEW

File: src/PhpOption/Some.php [hunk 1]
```diff
@@ -132,9 +134,9 @@ public function filterIsOneOf(...$classes)
             return None::create();
         }
 
-        if (is_array($classes[0]) || $classes[0] instanceof \Traversable) {
-            $classes = $classes[0];
-        }
+        $classes = new RecursiveIteratorIterator(
```
Comment (corporealshift) [inline src/PhpOption/Some.php:137]: it largely doesn't matter but the performance of this thing is really bad. I think we should instead just do 

`$result = array();
array_walk_recursive($original_array,function($v, $k) use (&$result){ $result[] = $v; });`

Comment (mister-spock) [reply src/PhpOption/Some.php:137]: I agree, but judging from tests, this bit of code can expect that classes are passed not only as plain arrays, but as ArrayIterator as well.
`array_walk_recursive` won't walk through that.

Comment (corporealshift) [reply src/PhpOption/Some.php:137]: i see. ok well the only thing then is you need to provide unit tests that cover the scenario you were seeing.

**Review by corporealshift — CHANGES_REQUESTED** [review]
Unit Tests

File: src/PhpOption/Some.php (hunk 1 above)
Comment (mister-spock) [reply src/PhpOption/Some.php:137]: There is one that uses `ArrayIterator`. I suppose the idea was to allow this to traverse over any `Traversable`.
I'll add some more.

File: tests/PhpOption/Tests/SomeTest.php [hunk 2]
```diff
@@ -105,10 +105,21 @@ public function testFilterIsOneOf()
 
         $this->assertSame($some, $some->filterIsOneOf(\stdClass::class, 'unknown'));
         $this->assertSame($some, $some->filterIsOneOf([\stdClass::class, 'unknown']));
-        $this->assertSame($some, $some->filterIsOneOf(new \ArrayIterator([\stdClass::class, 'unknown'])));
         $this->assertSame($some, $some->filterIsOneOf([[\stdClass::class, 'unknown']]));
         $this->assertSame($some, $some->filterIsOneOf([[[\stdClass::class, 'unknown']]]));
         $this->assertSame($some, $some->filterIsOneOf([[[[\stdClass::class, 'unknown']]]]));
+        $this->assertSame($some, $some->filterIsOneOf(new \ArrayIterator([\stdClass::class, 'unknown'])));
+        $this->assertSame($some, $some->filterIsOneOf([
```
Comment (corporealshift) [inline tests/PhpOption/Tests/SomeTest.php]: is this what you were trying to do? this use case isn't supported (nested arrays of types?) perhaps I was unclear:
I would like a unit test that will fail on the current version for the bug you are attempting to fix, that runs green with your fix.

**Review by corporealshift — CHANGES_REQUESTED** [review]

File: tests/PhpOption/Tests/SomeTest.php (hunk 2 above)
Comment (mister-spock) [reply tests/PhpOption/Tests/SomeTest.php]: Oh, I see. I'll add some.
The scenario is quite simple though.
You just need to create a `LazyOption` and then call `filterIsOneOf` with an array of classnames passed to it (comma-separated set of parameters works).

Comment (mister-spock) [reply tests/PhpOption/Tests/SomeTest.php]: Done. See `LazyOptionTest:: testFilterIsOneOf()`.
Last two assertions should be failing without my fix.

**Review by corporealshift — APPROVED** [review]

**Pull request merged**